File: Examples/Advertising/TargetedAdsDemo/BannerDemoViewController.m
//
//  BannerDemoViewController.m
//  Bazaarvoice Mobile Ads SDK - Demo Application
//
//  Copyright 2015 Bazaarvoice Inc. All rights reserved.
//

#import "BannerDemoViewController.h"

@implementation BannerDemoViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    

    self.bannerView.adUnitID = @"/6499/example/banner"; //Test adUnitId. Replace with your targeted adUnitId.
    self.bannerView.rootViewController = self;
    
    BVTargetedRequest* request = [self.bannerView getTargetedRequest];
    request.testDevices = @[ kGADSimulatorID ];
    [self.bannerView loadRequest:request];

    [self.view addSubview:self.bannerView];
}

- (IBAction)closeButtonPressed {
    [self.presentingViewController dismissViewControllerAnimated:YES completion:nil];
}

@end

File: Examples/Advertising/Swift/TargetedAdsDemo/BannerDemoViewController.swift
//
//  BannerDemoViewController.swift
//  TargetedAdsDemo
//
//  Copyright © 2016 Bazaarvoice. All rights reserved.
//

import UIKit
import GoogleMobileAds
import BVSDK

class BannerDemoViewController: UIViewController {

    @IBOutlet weak var closeButton: UIButton!
    @IBOutlet weak var bannerView: DFPBannerView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.title = "Banner Ad"
        
        // Request the banner ad
        self.bannerView.adUnitID = "/6499/example/banner" //Test adUnitId. Replace with your targeted adUnitId.
        self.bannerView.rootViewController = self;
        
        let request = DFPRequest()
        request.testDevices = [ kGADSimulatorID ]
        request.customTargeting = BVSDKManager.shared().getCustomTargeting()
        
        bannerView.load(request)
        self.view.addSubview(self.bannerView)
        
    }

    
    @IBAction func closeButtonPressed() {
        
        self.presentingViewController?.dismiss(animated: true, completion: nil)
        
    }

}
